examples/correlation_matrices/sampler.cpp

```diff
@@ -89,6 +89,26 @@ void write_to_file(std::string filename, std::vector<PointType> const& randPoint
     std::cout.rdbuf(coutbuf);
 }
 
+bool is_correlation_matrix(const MT& matrix){
+
+	const double tol = 1e-8;
+	
+	//check if all the diagonal elements are ones
+	for(int i=0 ; i<matrix.rows() ; i++){
+		if(std::abs(matrix(i, i)-1.0) > tol){
+			return false;
+		}
+	}
+	
+	//check if the matrix is positive definite
+	Eigen::SelfAdjointEigenSolver<MT> eigen_solver(matrix);
+	
+	if(eigen_solver.info() != Eigen::Success) return false;
+	
+	//the matrix is positive definite if all eigenvalues are positive
+	return eigen_solver.eigenvalues().minCoeff() > tol;
```

Comment:
I think it's better to check `return eigen_solver.eigenvalues().minCoeff() > -tol;`

Reply:
Okay yes, that sounds better! I'll incorporate this change